### PR TITLE
Singleton

### PR DESCRIPTION
**Pull request by patseng**
* Creates a Redis Model which is intended to be used as a Singleton
* Adds a SetField

### CODE REVIEW

File: rohm/models.py [hunk 1]
```diff
@@ -453,3 +455,25 @@ def __repr__(self):
 
     def __str__(self):
         return str(self._id)
+
+
+class Singleton(Model):
+    singleton_cache_key = None
```
Comment (alvinchow86) [inline rohm/models.py]: don't need the terminology `cache`, since rohm isn't necessarily a cache

Comment (alvinchow86) [inline rohm/models.py]: where is this being used? i don't see it set in the test either

File: tests/test_models.py [hunk 2]
```diff
@@ -299,3 +299,37 @@ def test_save_existing_raises_exception(Foo, pipe):
 
     foo2.save(force_create=True)
     assert Foo.get(id=1).name == 'foo2'
+
+@pytest.fixture
+def Bar():
+    class Bar(Singleton):
```
Comment (patseng) [inline tests/test_models.py]: @alvinchow86 here's where it's used

File: rohm/models.py (hunk 1 above)
Comment (alvinchow86) [inline rohm/models.py]: i mean the cache key

**Pull request closed without merging**